clients/java/src/main/java/marquez/client/models/Run.java

```diff
@@ -39,7 +43,9 @@ public Run(
       @Nullable final Instant endedAt,
       @Nullable final Long durationMs,
       @Nullable final Map<String, String> args,
-      @Nullable final Map<String, Object> facets) {
+      @Nullable final Map<String, Object> facets,
+      @Nullable final List<RunDatasetFacets> inputFacets,
+      @Nullable final List<RunDatasetFacets> outputFacets) {
```

Comment:
The class `Run` defines a list of `inputVersions` and `inputVersions` as `DatasetVersionId` entries. Given that a class `DatasetVersionId` is needed for `RunDatasetFacets` (along with the in/out facets) can we define a class `DatasetVersionIdAndFacets` as:

```java
class DatasetVersionIdAndFacets {
  final DatasetVersionId datasetVersionId;
  final ImmutableMap<String, Object> facets;
}
```

Then, update [`Run`](https://github.com/MarquezProject/marquez/blob/main/api/src/main/java/marquez/service/models/Run.java) to use `DatasetVersionIdAndFacets` for `inputVersions` and `outputVersions`?

```java
public class Run {
  .
  final List<DatasetVersionIdAndFacets> inputVersions;
  List<DatasetVersionIdAndFacets> outputVersions;
  .
  .
```

It just seems `RunDatasetFacets` is really a `DatasetVersionId` with facets?

Reply:
Mind also updating the [`spec/openapi.yml`](https://github.com/MarquezProject/marquez/blob/main/spec/openapi.yml) with the additional properties?

Reply:
@wslulciuc  I agree that merging these two entries is fine, but I would like to avoid `And` in class name. Could we stay within `RunDatasetFacets` which contain both dataset version id's and facets? I've modified the code to return Run in that form. 

Additionally, I've included changes in spec. 
![Screenshot 2023-03-03 at 09 29 36](https://user-images.githubusercontent.com/1646950/222671479-e2e366e8-344c-49a4-a47d-c0a606ba06cf.png)